Smallrye ConfigSourceFactory not loading class

### Describe the bug

When using a `ConfigSourceFactory` the class is not loaded and the build (both maven and gradle) fails with:

```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  2.737 s
[INFO] Finished at: 2021-11-25T11:24:15-03:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:2.5.0.Final:generate-code (default) on project bidder: Quarkus code generation phase has failed: InvocationTargetException: io.smallrye.config.ConfigSourceFactory: Provider com.adquick.service.athena.ConsumerConfig not found -> [Help 1]
```

Goals tested: `quarkus:dev`, `compile`, `package`.
The class referenced in the `src/main/resources/META-INF/services/io.smallrye.config.ConfigSourceFactory` file is being loaded before its compilation. 
Works when using Quarkus 1.13.4. 

### Expected behavior

The config factory should be loaded and executed. 

### Actual behavior

Buidl failes with:

```
Caused by: java.util.ServiceConfigurationError: io.smallrye.config.ConfigSourceFactory: Provider com.adquick.service.athena.ConsumerConfig not found
    at java.util.ServiceLoader.fail (ServiceLoader.java:593)
    at java.util.ServiceLoader$LazyClassPathLookupIterator.nextProviderClass (ServiceLoader.java:1219)
    at java.util.ServiceLoader$LazyClassPathLookupIterator.hasNextService (ServiceLoader.java:1228)
    at java.util.ServiceLoader$LazyClassPathLookupIterator.hasNext (ServiceLoader.java:1273)
    at java.util.ServiceLoader$2.hasNext (ServiceLoader.java:1309)
    at java.util.ServiceLoader$3.hasNext (ServiceLoader.java:1393)
```

### How to Reproduce?

Run quarkus:dev to reproduce: 

```bash
./mvnw clean compile quarkus:dev
```

A  "workaround":

1. Comment the class in:

```bash 
src/main/resources/META-INF/services/io.smallrye.config.ConfigSourceFactory
```

2. Run `./mvnw compile`

3. Uncomment the class

4. Run:

```bash
./mvnw quarkus:dev
```
[bidder-maven.zip](https://github.com/quarkusio/quarkus/files/7603733/bidder-maven.zip)




### Output of `uname -a` or `ver`

Darwin Andre-MacBook-Pro.local 20.6.0 Darwin Kernel Version 20.6.0: Mon Aug 30 06:12:21 PDT 2021; root:xnu-7195.141.6~3/RELEASE_X86_64 x86_64

### Output of `java -version`

openjdk version "17.0.1" 2021-10-19 OpenJDK Runtime Environment GraalVM CE 21.3.0 (build 17.0.1+12-jvmci-21.3-b05) OpenJDK 64-Bit Server VM GraalVM CE 21.3.0 (build 17.0.1+12-jvmci-21.3-b05, mixed mode, sharing)

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.5.0.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.1 (05c21c65bdfed0f71a2f2ada8b84da59348c4c5d)

### Additional information

_No response_

--- a/core/deployment/src/main/java/io/quarkus/deployment/CodeGenerator.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/CodeGenerator.java
@@ -92,7 +92,7 @@ public class CodeGenerator {
 
             final PropertiesConfigSource pcs = new PropertiesConfigSource(properties, "Build system");
 
-            final SmallRyeConfig config = ConfigUtils.configBuilder(false, launchMode)
+            final SmallRyeConfig config = ConfigUtils.configBuilder(false, false, launchMode)
                     .withProfile(launchMode.getDefaultProfile())
                     .withSources(pcs)
                     .build();

